packages/loader/container-definitions/src/loader.ts

```diff
@@ -198,6 +198,19 @@ export interface ILoader extends IFluidRouter {
     rehydrateDetachedContainerFromSnapshot(snapshot: string): Promise<IContainer>;
 }
 
+export type ILoaderOptions = {
```

Comment:
this syntax is a bit odd, why not just add an interface { hotSwapContext?: boolean, [key: string | number]: any}

Reply:
This is invalid:
```typescript
interface ISomething {
  [key: string | number]: any;
}
```
because of the union type in the indexer key.

But this is allowed:
```typescript
type ISomething = {
  [key in string | number]: any;
};
```

So to combine them, it needs to be an intersection type. TypeScript will actually automatically fix this error for you.

Reply:
we do something similar with IFluidPackage, and elsewhere. is it just the number type that throws it off? i don't think we use number configs, so it can just be exclude